### ISSUE TITLE
No Streams found

### ISSUE THREAD

**Issue by azaim** [anxdpanic/plugin.video.youtube#145]
### Context
Since today, youtube addon giving No Streams found error.

* Add-on Version: 6.8.11 & 6.8.14
* Kodi Version: 18.9
* Kodi GUI Language: English
* Operating System: Libreelec
* Operating System Language: English.

------

### Expected Behavior

Yesterday, it was working perfectly fine.

------

### Current Behavior

No Streams found error, when you try to open a video.

------

### Steps to Reproduce

Please provide detailed steps for reproducing the issue.

1. Open any video.

------

### Log

Please include a complete [debug log](https://kodi.wiki/view/Log_file).

2021-06-18 16:54:23.907 T:1401942912   ERROR: Traceback (most recent call last):
2021-06-18 16:54:23.907 T:1401942912   ERROR:   File "/storage/.kodi/addons/plugin.video.youtube/resources/lib/youtube_plugin/youtube/helper/yt_play.py", line 50, in play_video
2021-06-18 16:54:23.908 T:1401942912   ERROR:     video_streams = client.get_video_streams(context, video_id)
2021-06-18 16:54:23.908 T:1401942912   ERROR:   File "/storage/.kodi/addons/plugin.video.youtube/resources/lib/youtube_plugin/youtube/client/youtube.py", line 104, in get_video_streams
2021-06-18 16:54:23.908 T:1401942912   ERROR:     video_streams = video_info.load_stream_infos(video_id)
2021-06-18 16:54:23.908 T:1401942912   ERROR:   File "/storage/.kodi/addons/plugin.video.youtube/resources/lib/youtube_plugin/youtube/helper/video_info.py", line 515, in load_stream_infos
2021-06-18 16:54:23.908 T:1401942912   ERROR:     return self._method_get_video_info(video_id)
2021-06-18 16:54:23.908 T:1401942912   ERROR:   File "/storage/.kodi/addons/plugin.video.youtube/resources/lib/youtube_plugin/youtube/helper/video_info.py", line 1040, in _method_get_video_info
2021-06-18 16:54:23.908 T:1401942912   ERROR:     raise YouTubeException('No streams found')
2021-06-18 16:54:23.909 T:1401942912   ERROR: YouTubeException: No streams found
2021-06-18 16:54:23.909 T:1401942912   ERROR: [plugin.video.youtube] None
2021-06-18 16:54:23.912 T:1936932880   ERROR: Playlist Player: skipping unplayable item: 0, path [plugin://plugin.video.youtube/play/?video_id=jcqpxNnhlic]

------

### Additional Information

I read the previous bug report (https://github.com/anxdpanic/plugin.video.youtube/issues/139) and upgrade my plugin version to 6.8.14 but nothing changed. 

------

**Comment by ghost** [comment]
I am having the same problem with Kodi 18.9 and plugin version 6.8.14
Any new ideas?

**Comment by kuene** [comment]
I have the Problem still with 6.8.14.
```
2021-06-18 13:45:16.429 T:2952159248  NOTICE: Starting Kodi (18.9). Platform: Linux ARM (Thumb) 32-bit
2021-06-18 13:45:16.429 T:2952159248  NOTICE: Using Release Kodi x32 build (version for Vero)
2021-06-18 13:45:16.429 T:2952159248  NOTICE: Kodi compiled 2020-11-27 by GCC 8.3.0 for Linux ARM (Thumb) 32-bit version 4.19.152 (267160)
```

```
2021-06-18 16:39:39.386 T:2952159248   ERROR: Playlist Player: skipping unplayable item: 0, path [plugin://plugin.video.youtube/play/?video_id=XGZ1IbK2_Vw]
2021-06-18 16:40:37.101 T:2794451168   ERROR: Traceback (most recent call last):
2021-06-18 16:40:37.101 T:2794451168   ERROR:   File "/home/osmc/.kodi/addons/plugin.video.youtube/resources/lib/youtube_plugin/youtube/helper/yt_play.py", line 50, in play_video
2021-06-18 16:40:37.101 T:2794451168   ERROR:     video_streams = client.get_video_streams(context, video_id)
2021-06-18 16:40:37.102 T:2794451168   ERROR:   File "/home/osmc/.kodi/addons/plugin.video.youtube/resources/lib/youtube_plugin/youtube/client/youtube.py", line 104, in get_video_streams
2021-06-18 16:40:37.102 T:2794451168   ERROR:     video_streams = video_info.load_stream_infos(video_id)
2021-06-18 16:40:37.102 T:2794451168   ERROR:   File "/home/osmc/.kodi/addons/plugin.video.youtube/resources/lib/youtube_plugin/youtube/helper/video_info.py", line 515, in load_stream_infos
2021-06-18 16:40:37.103 T:2794451168   ERROR:     return self._method_get_video_info(video_id)
2021-06-18 16:40:37.103 T:2794451168   ERROR:   File "/home/osmc/.kodi/addons/plugin.video.youtube/resources/lib/youtube_plugin/youtube/helper/video_info.py", line 1040, in _method_get_video_info
2021-06-18 16:40:37.103 T:2794451168   ERROR:     raise YouTubeException('No streams found')
2021-06-18 16:40:37.103 T:2794451168   ERROR: YouTubeException: No streams found
2021-06-18 16:40:37.104 T:2794451168   ERROR: [plugin.video.youtube] None
2021-06-18 16:40:37.136 T:2952159248   ERROR: Playlist Player: skipping unplayable item: 0, path [plugin://plugin.video.youtube/play/?video_id=XGZ1IbK2_Vw]
```

**Comment by dardanel78** [comment]
That's how I solved the problem. Old API project completely deleted. New Youtube Data API V3 created. Create API key, create Oauth client ID.
Under Service Accounts you have to confirm an email. Then at OAuth consent screen, make internal test users + add users and enter your email address. Now you enter your API key, API ID, API secret in Kodi youtube. Then log in to Kodi youtube and log in to google.com/device with the generated code and it works

**Comment by azaim** [comment]
> That's how I solved the problem. Old API project completely deleted. New Youtube Data API V3 created. Create API key, create Oauth client ID.
> Under Service Accounts you have to confirm an email. Then at OAuth consent screen, make internal test users + add users and enter your email address. Now you enter your API key, API ID, API secret in Kodi youtube. Then log in to Kodi youtube and log in to google.com/device with the generated code and it works

I did it and it worked. Thank you @dardanel78

**Comment by pelztier3011** [comment]
yes, this work!

> That's how I solved the problem. Old API project completely deleted. New Youtube Data API V3 created. Create API key, create Oauth client ID.
> Under Service Accounts you have to confirm an email. Then at OAuth consent screen, make internal test users + add users and enter your email address. Now you enter your API key, API ID, API secret in Kodi youtube. Then log in to Kodi youtube and log in to google.com/device with the generated code and it works

yes! thx, this work on raspberry and win 10.

**Comment by tiboutchu** [comment]
Yes i have the same problem, and it's ok when i reconnect, but then, now, i have to reconnect (google.com/device) each time i reboot my kodi. This was not required before today.

**Comment by dardanel78** [comment]
> Yes i have the same problem, and it's ok when i reconnect, but then, now, i have to reconnect (google.com/device) each time i reboot my kodi. This was not required before today.

I can not confirm. The connection remains permanent after logging in twice. Even after the reboot

**Comment by Doktor-X** [comment]
I some how manage to folow to me unclear dardanel78 instructions (but i dont like this system where i need to be log in to youtube since i want watch videos withot login) but its works until proper fix is intreduce

I think that problem is with age restricted content

**Comment by PanedBeda** [comment]
> That's how I solved the problem. Old API project completely deleted. New Youtube Data API V3 created. Create API key, create Oauth client ID.
> Under Service Accounts you have to confirm an email. Then at OAuth consent screen, make internal test users + add users and enter your email address. Now you enter your API key, API ID, API secret in Kodi youtube. Then log in to Kodi youtube and log in to google.com/device with the generated code and it works

helped me out. Thy. 

i already thought, that it might be something that google changed.. 
tried to reinstall everything and so on.. and it is just a simple login issue that is not displayed..

**Comment by Doktor-X** [comment]
Try to open some videos that are age restricted, i have verify my age on same account that is login in kodi addon and in android youtube app, and on android app i can see videos without problems but on kodi addon i get "stream not found" and its only happening on videos that are marked as age restricted. Im sending videos from my youtube app to my kodi using yatse in order to se videos on big screen

**Comment by forgerpl** [comment]
I've encountered this bug on properly configured API v3. Deleting caches and reauthenticating (maintenance section in the settings) helped.

**Comment by ChibiChiii** [comment]
> 
> 
> That's how I solved the problem. Old API project completely deleted. New Youtube Data API V3 created. Create API key, create Oauth client ID.
> Under Service Accounts you have to confirm an email. Then at OAuth consent screen, make internal test users + add users and enter your email address. Now you enter your API key, API ID, API secret in Kodi youtube. Then log in to Kodi youtube and log in to google.com/device with the generated code and it works

Awesome, this did indeed do the trick.
I deleted the old API key entirely from my Google API Console and created a new one per the instructions here: https://seo-michael.co.uk/how-to-create-your-own-youtube-api-key-id-and-secret/

Then added myself as a test user, and logged into Youtube after entering the new API key in the Kodi Youtube Addon. Now sharing videos to Kodi works again! :D Thank you so much.

**Comment by Web-Eye** [comment]
> That's how I solved the problem. Old API project completely deleted. New Youtube Data API V3 created. Create API key, create Oauth client ID.
> Under Service Accounts you have to confirm an email. Then at OAuth consent screen, make internal test users + add users and enter your email address. Now you enter your API key, API ID, API secret in Kodi youtube. Then log in to Kodi youtube and log in to google.com/device with the generated code and it works

seems to work, but how complicated, and have we to create api keys every time, if google change anything?

**Comment by Doktor-X** [comment]
I have v3 api from 2020 and only thing that i added is this extra step of adding new test user and login with that same user (thing that i did not needed befor, i have always used youtube without login) and i cant say that this is not working now, its working but its not working for age verify content

**Comment by betatester3016** [comment]
This seems to work:

      Sign in and reauthenticate with a valid tied Android device.
      Once you press the sign in button it throws up a code.
      Go to your Android device, navigate to google.com/device and enter the code.
      It will probably complain about HTML been insecure so you have to accept an insecure connection to proceed.

**Comment by rafhtl** [comment]
https://github.com/anxdpanic/plugin.video.youtube/pull/153

this works for me without having to change anything in api
thank you

**Comment by anxdpanic** [comment]
https://github.com/anxdpanic/plugin.video.youtube/releases/tag/6.8.15

**Comment by Doktor-X** [comment]
> https://github.com/anxdpanic/plugin.video.youtube/releases/tag/6.8.15

And now how to make age videos play? Im login on addon with user that on android app have age restrictions removed.

**Comment by anxdpanic** [comment]
age gated videos are working for me when logged in. would need the log (in a new issue) to hopefully see something, if that's not the case for you

**Comment by XDoKToR** [comment]
You gotta love  anxdpanic for these fixes...LEGEND!

and thanks for sort of remixing my name 'Doktor-X' lol

**Comment by asavah** [comment]
> https://github.com/anxdpanic/plugin.video.youtube/releases/tag/6.8.15

Hmm so it looks like I was right about "consent" cookie stuff :)

Thanks much @anxdpanic 
/me bows

**Comment by azaim** [comment]
> https://github.com/anxdpanic/plugin.video.youtube/releases/tag/6.8.15

Thank you anxdpanic.

**Comment by jmbreuer** [comment]
I came across this because I was having "Video restricted" playback issues, despite being logged in with a verified account.

To fix the problem, I had to create new Google OAuth Client Credentials, using the "TV or limited input" device type.

That option was not available when I'd created those credentials originally, and apparently it's not something one can change on an existing credential.

I followed only the OAuth Client ID part of this recipe (referenced above):

https://seo-michael.co.uk/how-to-create-your-own-youtube-api-key-id-and-secret/

Slight caveat: The URL given there for the developer console is wrong/outdated and leads to a "your developer account is temporarily locked" error page.

https://console.cloud.google.com/

This appears to be a currently valid URL for the Google Developer Console, and it's also possible to navigate there from the bottom rider of https://developers.google.com
